app/assets/javascripts/channels/comments.js

```diff
@@ -24,7 +24,7 @@ if ($("meta[name='push_notifications']").length >0) {
         channel: "CommentsChannel",
         notification: $(location).attr('href').split('/').pop()},{
         received: function(data){
-          if ($('#notification-thread').attr('data-id') == data.subject_id){
+          if ($('#notification-thread').attr('data-id') == data.subject_id && !$("#comment-"+data.comment_id).length){
             $('.discussion-thread').append(data.comment_html);
```

Comment:
Looks like you've got the same code here twice, but the existing comment check only in the second block? Can you pull the logic out into a function to reduce the duplication?

Reply:
exiting comment being shown would only happen if we had just posted a comment (which kick a turbolink load). App.comments is re-initialised on each of these, so if I attach it there the code is in two places again. If I attach it to Octobox then the code is in another place which seems a bit weird. Not sure where else to drop it.

Reply:
I'd prefer to reduce the duplication to avoid future copy-pasta bugs

Reply:
not questioning that strategy, just not sure where to attach that code.